build/build.go

```diff
@@ -454,6 +457,7 @@ func toSolveOpt(d driver.Driver, multiDriver bool, opt Options, dl dockerLoadCal
 	switch opt.NetworkMode {
 	case "host", "none":
 		so.FrontendAttrs["force-network-mode"] = opt.NetworkMode
+		so.AllowedEntitlements = append(so.AllowedEntitlements, entitlements.EntitlementNetworkHost)
```

Comment:
nit: should check whether so.AllowedEntitlements already contains EntitlementNetworkHost?

Reply:
It doesn't matter for the builder, but if you want I can clean it up.